### PR TITLE
Arabic language support

### PR DESCRIPTION
**Pull request by AbdallahAHO**
Arabic Language (Modern Standard Arabic - Al-fussha) support for date-fns.

### CODE REVIEW

File: src/locale/ar/build_distance_in_words_locale/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      other: 'حوالي {{count}} أشهر'
+    },
+
+    xMonths: {
+      one: 'شهر واحد',
+      other: '{{count}} أشهر'
+    },
+
+    aboutXYears: {
+      one: 'عام واحد تقريباً',
```
Comment (ahmedelgabri) [inline src/locale/ar/build_distance_in_words_locale/index.js]: I see here that you are using تقريبا while up you are using حوالي, I guess تقريبا is the best for all of them

File: src/locale/ar/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+var buildDistanceInWordsLocale = require('./build_distance_in_words_locale/index.js')
+var buildFormatLocale = require('./build_format_locale/index.js')
+
+/**
+ * @category Locales
+ * @summary English locale.
```
Comment (ahmedelgabri) [inline src/locale/ar/index.js]: Shouldn't this be Arabic? :)

Comment (AbdallahAHO) [inline src/locale/ar/index.js]: Missed this! :sweat_smile:

File: src/locale/ar/build_distance_in_words_locale/index.js (hunk 1 above)
Comment (AbdallahAHO) [inline src/locale/ar/build_distance_in_words_locale/index.js]: Good point, Changed :+1:

File: src/locale/ar/build_format_locale/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  })
+
+  return {
+    formatters: formatters,
+    formattingTokensRegExp: buildFormattingTokensRegExp(formatters)
+  }
+}
+
+// Arabic Language doesn't have fixed prefix for ordinal Numbers similar to "st", "nd", "rd" or "th"
+// We Will need
```
Comment (LeshaKoss) [inline src/locale/ar/build_format_locale/index.js]: What is the problem with ordinal numbers? It's ok to return just numbers if there are none, we do it in Icelandic locale IIRC. If so, please remove this comment

Comment (ahmedelgabri) [inline src/locale/ar/build_format_locale/index.js]: @LeshaKoss I agree, makes more sense. As far as I know, ordinals are only used in formatting, right? https://date-fns.org/docs/format if this is the only case then I think numbers will be fine.

Comment (LeshaKoss) [inline src/locale/ar/build_format_locale/index.js]: @ahmedelgabri yes, the locale returned by buildFormatLocale is used only in `format` ATM

File: src/locale/ar/build_format_locale/index.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+function buildFormatLocale () {
+  // Note: in English, the names of days of the week and months are capitalized.
+  // If you are making a new locale based on this one, check if the same is true for the language you're working on.
+  // Generally, formatted dates should look like they are in the middle of a sentence,
+  // e.g. in Spanish language the weekdays and months should be in the lowercase.
+  var months3char = ['يناير', 'فبراير', 'مارس', 'أبريل', 'مايو', 'يونيو', 'يوليو', 'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر']
+  var monthsFull = ['كانون الثاني يناير', 'شباط فبراير', 'آذار مارس', 'نيسان أبريل', 'أيار مايو', 'حزيران يونيو', 'تموز يوليو', 'آب أغسطس', 'أيلول سبتمبر', 'تشرين الأول أكتوبر', 'تشرين الثاني نوفمبر', 'كانون الأول ديسمبر']
+  var weekdays2char = ['أح', 'إث', 'ثل', 'أر', 'خم', 'جم', 'ست']
+  var weekdays3char = ['أحد', 'إثن', 'ثلاث', 'أرب', 'خمس', 'جمع', 'سبت']
```
Comment (ahmedelgabri) [inline src/locale/ar/build_format_locale/index.js]: As far as I know in Arabic there is no 2 or 3char for days & for me it looks odd. If you check moment for example you will see that there is no 2 or 3 they just drop "ال" for the short version. & for the very short they use one letter.

https://github.com/moment/moment/blob/develop/locale/ar.js#L79-L80

This is even the case with nearly all Arabic dialects too
https://github.com/moment/moment/tree/develop/locale

For Egypt for example the 2char for Sunday اح won't sound nice at all.

Comment (LeshaKoss) [inline src/locale/ar/build_format_locale/index.js]: @ahmedelgabri @AbdallahAHO it's ok to make it not 2 or 3 characters but just the shortest and a short abbreviation of the month/the day of the week. We do this in a lot of locales. I think it's 👍

Comment (AbdallahAHO) [inline src/locale/ar/build_format_locale/index.js]: @ahmedelgabri, I agree with the weird أح, I was more into using moment.js format at first, but I wasn't sure if it's okay to make it more than 2 or 3 characters. However, It's fixed it now. :+1:

File: src/locale/ar/build_format_locale/index.js (hunk 3 above)
Comment (AbdallahAHO) [inline src/locale/ar/build_format_locale/index.js]: Comment removed! :+1:

**Comment by LeshaKoss** [comment]
@AbdallahAHO @ahmedelgabri looks good! Is it ready to merge?

**Comment by ahmedelgabri** [comment]
@LeshaKoss it's ok with me to be merged yes :)

**Comment by LeshaKoss** [comment]
@AbdallahAHO thanks a lot for your awesome contribution! ⭐️ Arabic locale was released as [v1.24.0](https://github.com/date-fns/date-fns/releases/tag/v1.24.0) (diff: https://github.com/date-fns/date-fns/compare/v1.23.0...v1.24.0)

**Comment by AbdallahAHO** [comment]
That's my first contribution to a public repo! :tada: 
Thank you @LeshaKoss for making date-fns possible, and Thanks for your help @ahmedelgabri, I am a big fan of your work.

**Pull request merged**